src/System.Text.Json/tests/Serialization/Array.ReadTests.cs

```diff
@@ -78,6 +78,15 @@ public static void ReadByteArrayFail()
             Assert.Throws<JsonException>(() => JsonSerializer.Parse<byte[]>(@"""A==="""));
         }
 
+        [Theory]
+        [InlineData(typeof(int[]), @"[{}]")]
+        [InlineData(typeof(Dictionary<string, int[]>), @"{""test"": [{}]}")]
+        public static void InvalidJsonForArrayShouldFail(Type type, string json)
+        {
+            // These fail because the int converter sees the StartObject token.
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse(json, type));
+        }
+
         [Fact]
         public static void ReadByteArrayAsJsonArrayFail()
         {
```

Comment:
Unrelated to your change, this test looks the same as the ReadByteArrayFail test above, just with different inputs.  Can we just make that a theory? e.g.
```C#
[Theory]
[InlineData(@"""1""")]
[InlineData(@"""A===""")]
[InlineData("[1, 2]")]
public static void Parse_InvalidJsonArray_Throws(string json)
{
    Assert.Throws<JsonException>(() => JsonSerializer.Parse<byte[]>(json))
}
```
?